File: src/main/java/com/naminhyeok/fantazzk/template/TemplateCatalogException.java
package com.naminhyeok.fantazzk.template;

import java.util.UUID;

public sealed class TemplateCatalogException extends RuntimeException permits TemplateCatalogException.NotFound {
    private TemplateCatalogException(String message) {
        super(message);
    }

    public static final class NotFound extends TemplateCatalogException {
        public NotFound(UUID templateId) {
            super("템플릿을 찾을 수 없습니다: " + templateId);
        }
    }
}

File: src/main/kotlin/com/naminhyeok/fantazzk/template/TemplateCatalogException.kt
package com.naminhyeok.fantazzk.template

sealed class TemplateCatalogException(
    val templateId: TemplateId,
    message: String,
) : RuntimeException(message) {
    class NotFound(
        templateId: TemplateId,
    ) : TemplateCatalogException(templateId, "템플릿을 찾을 수 없습니다")

    class Invalid(
        templateId: TemplateId,
    ) : TemplateCatalogException(templateId, "유효하지 않은 템플릿입니다")
}
